compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/ClassReference.kt

```diff
@@ -77,6 +79,8 @@ public sealed class ClassReference {
   public abstract fun isInterface(): Boolean
   public abstract fun isAbstract(): Boolean
   public abstract fun visibility(): Visibility
+  public abstract fun boundTypeOrNull(annotationFqName: FqName): FqName?
```

Comment:
This function needs to be moved into `AnnotationReference`. An annotation has a bound type, not a class. This is one of the utilities that needs to be refactored more.

Reply:
Not sure what level of additional refactoring you were thinking, I went ahead and moved this to `AnnotationReference` and updated usages to find the annotation they want first, then call `boundTypeOrNull` on it. Let me know if you had additional refactoring in mind.

Reply:
No, I meant exactly what you did 👍🏻